services/src/main/java/org/keycloak/vault/PlainTextVaultProviderFactory.java

```diff
@@ -0,0 +1,76 @@
+package org.keycloak.vault;
+
+import org.jboss.logging.Logger;
+import org.keycloak.Config;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Creates and configures {@link PlainTextVaultProvider}.
+ *
+ * @author Sebastian Łaskawiec
+ */
+public class PlainTextVaultProviderFactory implements VaultProviderFactory {
+
+    private static final Logger logger = Logger.getLogger(MethodHandles.lookup().lookupClass());
+
+    public static final String PROVIDER_ID = "plaintext";
+
+    private String vaultDirectory;
+    private boolean disabled;
+    private Path vaultPath;
+
+    @Override
+    public VaultProvider create(KeycloakSession session) {
+        if (disabled || vaultDirectory == null) {
+            //init method not called?
+            throw new IllegalStateException("Can not create a vault since it's disabled or not initialized correctly");
+        }
+        return new PlainTextVaultProvider(vaultPath, session.getContext().getRealm().getName());
+    }
+
+    @Override
+    public void init(Config.Scope config) {
+        vaultDirectory = config.get("dir");
+
+        Boolean disabledFromConfig = config.getBoolean("disabled");
+        if (disabledFromConfig == null) {
```

Comment:
This can be null apparently, Travis fails due to this. Treat `disabled` as set if to `false` when `null`

Reply:
Good point.